app/models/firmware_registry_simple.rb

```diff
@@ -0,0 +1,59 @@
+class FirmwareRegistrySimple < FirmwareRegistry
+  def sync_fw_binaries_raw
+    remote_binaries.each do |binary_hash|
+      binary = FirmwareBinary.find_or_create_by(:firmware_registry => self, :external_ref => binary_hash['id'])
+      _log.info("Updating FirmwareBinary [#{binary.id} | #{binary.name}]...")
+
+      binary.name = binary_hash['filename'] || binary_hash['id']
+      binary.description = binary_hash['description']
+      binary.version = binary_hash['version']
+      binary.save!
+
+      unless binary.urls.sort == binary_hash['urls'].sort
+        _log.info("Updating FirmwareBinary [#{binary.id} | #{binary.name}] endpoints...")
+        binary.endpoints.destroy_all
```

Comment:
This could be a bit better if we consider what needs to be deleted and what needs to be added,

```
endpoints_by_url = binary.endpoints.index_by(&:url)
urls_to_delete = binary.urls - binary_hash['urls']
urls_to_delete.each { |url| endpoints_by_url[url].destroy }
urls_to_create = binary_hash['urls'] - binary.urls
urls_to_create.each { |url| binary.endpoints.create!(:url => url) }
```

Reply:
Sweet, thanks, integrated. I assumed that binary endpoints are so rarely updated that we don't care performance on such event, but it can't hurt to do it properly.